### DISCUSSION TITLE
Why have Windows ARM64 releases stopped?

### DISCUSSION THREAD

**Discussion by infinitepower18** [python-pillow/Pillow#7547] [Q&A] [answered]
I've noticed no more Windows ARM64 wheels are being published. The last version to get one is 10.0.0. So my ARM64 machine attempts to download the tar.gz and fails to install.

Why have the releases stopped? I find it strange considering Microsoft is investing a lot into Windows on ARM these days.

**Comment by radarhere** [comment, marked as answer]
As an explanation, the arm wheels were being [provided voluntarily by cgohlke](https://github.com/python-pillow/Pillow/issues/7195#issuecomment-1620987469), rather than being a standard part of our release process. He's not an official part of the Pillow team, and [his lab lost funding](https://github.com/python-pillow/Pillow/issues/6369#issue-1272298442), so that was just something that was happening out of the kindness of his heart.

GitHub Actions [doesn't provide ARM64 Windows environments](https://github.com/actions/runner-images/issues/768), so we can't build the wheels for it as easily as we can the other Windows wheels.

Just to be clear, you can [install Pillow from source](https://pillow.readthedocs.io/en/stable/installation.html#building-from-source), it is just not as convenient as using pre-build wheels.

**Reply by hugovk** [reply]
It's possible to build Windows ARM64 wheels on GitHub Actions via cross-compilation using cibuildwheel. 

https://github.com/pypa/cibuildwheel#usage

But we're using multibuild and I don't think multibuild supports it? 

I would like us to move over to cibuildwheel at some point.

**Reply by radarhere** [reply]
I've created https://github.com/python-pillow/Pillow/pull/7552 as a first step in this.

**Comment by nulano** [comment]
In case it is helpful, I have compiled Pillow 10.1.0 on an M2 MacBook in a Windows ARM64 virtual machine:

[Pillow-10.1.0-cp39-cp39-win_arm64.zip](https://github.com/python-pillow/Pillow/files/13383057/Pillow-10.1.0-cp39-cp39-win_arm64.zip)
[Pillow-10.1.0-cp312-cp312-win_arm64.zip](https://github.com/python-pillow/Pillow/files/13383058/Pillow-10.1.0-cp312-cp312-win_arm64.zip)
[Pillow-10.1.0-cp311-cp311-win_arm64.zip](https://github.com/python-pillow/Pillow/files/13383059/Pillow-10.1.0-cp311-cp311-win_arm64.zip)
[Pillow-10.1.0-cp310-cp310-win_arm64.zip](https://github.com/python-pillow/Pillow/files/13383061/Pillow-10.1.0-cp310-cp310-win_arm64.zip)
[fribidi_arm64.dll.zip](https://github.com/python-pillow/Pillow/files/13383051/fribidi_arm64.dll.zip)
